File: src/main/java/ru/arc/stock/HistoryMessager.java
package ru.arc.stock;

import java.util.Map;

import com.google.gson.reflect.TypeToken;
import lombok.RequiredArgsConstructor;
import ru.arc.ARC;
import ru.arc.network.ChannelListener;
import ru.arc.network.RedisManager;
import ru.arc.util.Common;

import static ru.arc.util.Logging.error;

@RequiredArgsConstructor
public class HistoryMessager implements ChannelListener {
    public final String channel;
    private final RedisManager redisManager;

    @Override
    public void consume(String channel, String message, String originServer) {
        if (ARC.serverName.equalsIgnoreCase(originServer)) return;
        try {
            TypeToken<Map<String, HistoryManager.HighLow>> typeToken = new TypeToken<>() {
            };
            Map<String, HistoryManager.HighLow> highLowMap = Common.gson.fromJson(message, typeToken.getType());
            HistoryManager.setHighLows(highLowMap);
        } catch (Exception e) {
            error("Error consuming highlows", e);
        }
    }

    public void send(Map<String, HistoryManager.HighLow> highLowMap) {
        try {
            String json = Common.gson.toJson(highLowMap);
            redisManager.publish(channel, json);
        } catch (Exception e) {
            error("Error sending highlows", e);
        }
    }
}

File: src/main/kotlin/ru/arc/stock/HistoryMessager.kt
package ru.arc.stock

import com.google.gson.reflect.TypeToken
import ru.arc.ARC
import ru.arc.network.ChannelListener
import ru.arc.network.RedisManager
import ru.arc.util.Common
import ru.arc.util.Logging.error

class HistoryMessager(
    val channel: String,
    private val redisManager: RedisManager,
) : ChannelListener {

    override fun consume(channel: String, message: String, originServer: String) {
        if (ARC.serverName.equals(originServer, ignoreCase = true)) return
        try {
            val typeToken = object : TypeToken<Map<String, HistoryManager.HighLow>>() {}
            val highLowMap: Map<String, HistoryManager.HighLow> = Common.gson.fromJson(message, typeToken.type)
            HistoryManager.setHighLows(highLowMap)
        } catch (e: Exception) {
            error("Error consuming highlows", e)
        }
    }

    fun send(highLowMap: Map<String, HistoryManager.HighLow>) {
        try {
            redisManager.publish(channel, Common.gson.toJson(highLowMap))
        } catch (e: Exception) {
            error("Error sending highlows", e)
        }
    }
}
